Libplanet/Blockchain/BlockChain.cs

```diff
@@ -464,7 +455,13 @@ public Transaction<T> GetTransaction(TxId txId)
             _rwlock.EnterReadLock();
             try
             {
-                return _transactions[txId];
+                if (Store.GetTransaction<T>(txId) is { } transaction)
```

Comment:
How about removing `GetTransaction` from `BlockChain<T>`?

Reply:
If it's unused by other projects (e.g., Lib9c, NineChronicles.Headless) that would be good.

Reply:
It's still used by `Swarm<T>`:

https://github.com/planetarium/libplanet/blob/ebce8116322b5cada571aa0055dde601b48f4f77/Libplanet/Net/Swarm.MessageHandlers.cs#L168

Reply:
IMO replace it to `IStore.GetTransaction` looks better, for unity.

Reply:
> IMO replace it to `IStore.GetTransaction` looks better, for unity.

@limebell Could you elaborate?

IMHO the methods of `IStore` are `BlockChain<T>`'s implementations details that should not be exposed on its surface.  If game apps use (= need) `IStore` directly, I'll find it [design smell][1], more exactly, *missing abstraction*.

On the other hand, if game app use (= need) neither `IStore.GetTransaction<T>()` nor `BlockChain<T>.GetTransaction()`, it means there's no demand for `BlockChain<T>.GetTransaction()` so that having this method is *deficient encapsulation*.

Anyway, I try to look into other projects that use Libplanet… and I found several demands in reality:

https://github.com/planetarium/lib9c/blob/614ab68915f14f0bcf933f06afe629ad0af1ff03/Lib9c/Miner.cs#L51
https://github.com/planetarium/lib9c/blob/614ab68915f14f0bcf933f06afe629ad0af1ff03/Lib9c/Miner.cs#L62
https://github.com/planetarium/lib9c/blob/614ab68915f14f0bcf933f06afe629ad0af1ff03/Lib9c/ReorgMiner.cs#L73
https://github.com/planetarium/lib9c/blob/614ab68915f14f0bcf933f06afe629ad0af1ff03/Lib9c/ReorgMiner.cs#L84

One more example, but it's not open sourced (yet):

https://github.com/planetarium/nekoyume-unity/blob/cc3d8a82d60f3fe04a4da0cb6f5e396f34c4e093/nekoyume/Assets/_Scripts/BlockChain/Agent.cs#L190

[1]: https://en.wikipedia.org/wiki/Design_smell